### PR TITLE
Fix: Copy/Alias Page also copies restrictions to the copied page

### LINKED ISSUE #10111: `public()` filter method does not remove alias pages of private pages from queryset
### Issue Summary

The `public()` filter method of the page queryset does not remove alias pages of private pages from the queryset.

### Steps to Reproduce

1. Checkout the latest version of the bakerydemo. 
2. Edit the `get_context` method of the `BlogIndex` in `bakerydemo/blog/models.py` to only show public pages, by adding the `.public()` filter method:
   ```python
    def get_context(self, request):
        context = super(BlogIndexPage, self).get_context(request)
        context["posts"] = (
            BlogPage.objects.descendant_of(self).live().order_by("-date_published").public()
        )
        return context

   ```
3. Have a look at what is listed on the blog index: http://localhost:8000/blog/
4. Set one of the pages to be private, e.g. by defining a password: http://localhost:8000/admin/pages/77/edit/
5. Check the index again to see that the page you set to private is not displayed anymore. 
6. Now create an alias copy of the page you set to private: http://localhost:8000/admin/pages/77/copy/?next=%2Fadmin%2Fpages%2F61%2F
7. Check the blog index and see that the alias page is listed on the index. 
8. Try to visit the alias page and see that it is protected like the original page (I have added `alias-` to the beginning of the slug when creating it): http://localhost:8000/blog/alias-desserts-benefits/

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

- I would expect the alias page to not be listed in the index because the index uses the `public()` filter. 
- I first ran into this in the context of translated pages, which are first created as alias pages. 
- I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: (yes / ~~no~~) With bakerydemo. 

### Technical details

-   Python version: 3.9.9
-   Django version: 4.1.7
-   Wagtail version: 4.2 (discovered on 2.16)

### PR DESCRIPTION
**Pull request by GeekGawd**
Fixes #10111 

- Add functionality to copy page restrictions if the page is copied or aliased.
- Prevent copying the restrictions if the parent page has restrictions.
- Add test cases in test_page_model.py to check if the restrictions were copied correctly in case parent has or doesn't have restrictions

_Please check the following:_

-   [x] Do the tests still pass?[^1]
-   [x] Does the code comply with the style guide?
    -   [x] Run `make lint` from the Wagtail root.
-   [x] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [ ] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**.
With tests I have essentially tried to cover Four Cases

- copy: when parent_page has restrictions/when parent_page doesn't have restrictions
- alias: when parent_page has restrictions/when parent_page doesn't have restrictions

### CODE REVIEW

**Comment by GeekGawd** [comment]
@lb- Sir, could you review this pr, I think it has gone under your sights. Thank You!

**Comment by lb-** [comment]
@GeekGawd it's flagged for a review - I'm not sure I'm the best person to review this sorry. Others in the core team should see this comment and review when they get the cha chance.

**Comment by lb-** [comment]
@zerolab do you mind giving this one a review when you get the chance?

**Comment by zerolab** [comment]
@lb- can try to have a look over the weekend (that is the earliest, I'm afraid)

**Comment by lb-** [comment]
Thanks Dan. I'll add you as a reviewer for now.

**Comment by GeekGawd** [comment]
hi! @zerolab sir, could you please give this pr a review when you find the time.

**Comment by zerolab** [comment]
I should have some dedicated time from next week, so will look at reviewing it then

File: wagtail/actions/copy_page.py [hunk 1]
```diff
@@ -326,6 +325,27 @@ def _copy_page(
                 page_copy.numchild = numchild
                 page_copy.save(clean=False, update_fields=["numchild"])
 
+        if to:
```
[the remark this answers is not in the document]
Comment (lb-) [reply wagtail/actions/copy_page.py:328]: @GeekGawd were you able to look at this comment?

**Comment by lb-** [comment]
Pinging for another look if you do get a chance @zerolab

**Comment by lb-** [comment]
Sorry Dan - did not mean to spam you, accidentally removed your review request.

**Comment by lb-** [comment]
A potentially relevant PR - copying privacy settings when copying pages. https://github.com/wagtail/wagtail/pull/6406

File: wagtail/actions/copy_page.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (gasman) [reply wagtail/actions/copy_page.py:328]: I was mistaken here - we do want to consider view restrictions set on the _destination_ parent (including ones inherited from ancestors), and if there are any, don't copy the source page's restrictions across (as that would result in multiple restrictions on the resulting page).

**Comment by gasman** [comment]
Thanks @GeekGawd! I made a few changes when merging #6406 that conflicted with this one, so I rebuilt this PR at #10801 and have now merged.

**Pull request closed without merging**